doc/example-keywords.py

```diff
@@ -0,0 +1,211 @@
+    match = re.search(fr"(?:{comment_char}.*?\n)+", text,
+                      re.DOTALL | re.MULTILINE | re.IGNORECASE)
+    global EXIT_CODE
+    if not match:
+        EXIT_CODE = 1
+        logging.error(f"Couldn't parse docstring for {filename}")
+        return False
+    docstring = match.group(0) + "\n\n"
+    docstring = "\n".join(line.lstrip(f"{comment_char} ")
+                          for line in docstring.split("\n"))
+
+    match = re.search(r"\s*Keywords:(.*?)\n\n", docstring, re.DOTALL | re.MULTILINE)
+    if not match:
+        EXIT_CODE = 1
+        logging.warning(f"No keywords found in {filename}")
+        return False
+    keywords = set(kw.strip() for kw in match[1].split(","))
+    return keywords
+
+
+def get_all_keywords():
+    """
+    Read keywords from all Cantera examples and print out a summary list
+    """
+    text = (Path(__file__).parent / "example-skip-keywords.txt").read_text()
+    skip = set(text.split("\n"))
+
+    all_keywords = Counter()
+    for d in Path("interfaces/cython/cantera/examples").glob("**"):
+        if d.is_dir():
+            for f in d.glob("*.py"):
+                if f.name.startswith("_") or f.name in skip:
+                    continue
+                kw = get_python_keywords(f)
+                if kw:
+                    all_keywords.update(kw)
+
+    for f in Path("samples/matlab").glob("*.m"):
+        if f.name in skip:
+            continue
+        kw = get_matlab_keywords(f)
+        if kw:
+            all_keywords.update(kw)
+
+    for d in Path("samples/cxx").glob("**"):
+        if f.name in skip:
+            continue
+        if d.is_dir():
+            for f in d.glob("*.cpp"):
+                all_keywords.update(get_cxx_keywords(f))
+
+    for f in Path("samples/f77").glob("*.f"):
+        if f.name in skip:
+            continue
+        kw = get_fortran_keywords(f, "c")
+        if kw:
+            all_keywords.update(kw)
+
+    for f in Path("samples/f90").glob("*.f90"):
+        if f.name in skip:
+            continue
+        kw = get_fortran_keywords(f, "!")
+        if kw:
+            all_keywords.update(kw)
+
+    return all_keywords
+
+
+def compare():
+    """
+    Print a list of keywords appearing in examples that are not in the known keywords
+    list. Return True if there are any such items.
+    """
+    text = (Path(__file__).parent / "example-keywords.txt").read_text()
+    known = set(text.split("\n"))
+    current = set(get_all_keywords())
+    delta = current - known
+    for kw in delta:
+        logging.warning(f"Keyword {kw!r} not in known keywords list")
+    return len(delta) > 0
+
+
+def save_keywords():
+    """
+    Save an updated version of the known keywords list based on keywords appearing in
+    any of the examples.
+    """
+    found_kw = get_all_keywords()
+    with open(Path(__file__).parent / "example-keywords.txt", "w") as known:
+        known.writelines(kw + "\n" for kw in sorted(found_kw))
```

Comment:
```suggestion
    (Path(__file__).parent / "...").write_text("\n".join(sorted(found_kw)))
```
I think this is simpler, but YMMV 😊 might take 2 lines anyways so there might not be an improvement.

Reply:
Your example loses the newline at the end of the file, but I was able to fix that and fit it all on one line.